website/docs/reference/widgets/table/README.md

```diff
@@ -60,183 +114,231 @@ Now, the table widget should be populated with the data coming from the query.
 
-#### Total record count
+1. Enable the server-side pagination property in the table
+2. Call the API / query via the **onPageChange** event
+3. Configure pagination in the API / query using the [Offset](#offset-based-pagination) or [Key-based](#key-based-pagination) pagination method.
 
-Total record count stores the total number of rows in the table. It's useful in pagination as it helps in determining the number of pages, thus enabling/disabling the next/previous page control in the table. It's only visible when you enable **server-side pagination**.
+#### pageNo
+
+`pageNo` gets the page number of the table that the user is currently viewing. This property can be used by APIs for pagination. To use this property in a widget, enter the code snippet given below:
 
-<VideoEmbed host="youtube" videoId="p7mH00xp7Nc" title="Using Total Records Count in th Table Widget" caption="Using Total Records Count in th Table Widget"/>
+```javascript
+{{<table_name>.pageNo}}
+```
+
+The following video shows how to bind a text widget to `Table_1` using `pageNo`.
+
+<VideoEmbed host="youtube" videoId="DqKok2cCJk0" title="pageNo" caption="pageNo"/>
+
+#### pageOffset
+
+This value is calculated by multiplying the table's page number with the amount of records displayed per page. For example, if there are five records on each page, and the table is showing page three, then `Table1.pageOffset` should be equal to 10. Use this value in your query to fetch the correct set of results by asking it to skip the first 10 records.
+
+```javascript
+{{<table_name>.pageOffset}}
+```
+
+The following video shows how to use `pageOffset` in your API / database query:
+
+<VideoEmbed host="youtube" videoId="rPw2KRgq3xc" title="pageOffset" caption="pageOffset"/>
+
+#### pageSize
+
+`pageSize` shows the total number of rows displayed on a page of the table. `pageSize` can change upon resizing the table.
+To bind a widget using this property, enter the code snippet given below:
+
+```javascript
+{{<table_name>.pageSize}}
+```
 
-To get the Total record count of your data, follow the steps below:
+The following video shows how to bind a text widget to `Table_1` using pageSize.
+
+<VideoEmbed host="youtube" videoId="XkFJQh4vcCw" title="pageSize" caption="pageSize"/>
+
+#### Total records
+
+The total records field stores the total number of rows in the table. It's useful in pagination as it helps in determining the number of pages, and then when to turn on/off the page control buttons in the table header. It's only visible when you enable **Server Side Pagination**.
+
+<VideoEmbed host="youtube" videoId="p7mH00xp7Nc" title="Using Total Records in the Table Widget" caption="Using Total Records in the Table Widget"/>
+
+To get the total record count of your data, follow the steps below:
```

Comment:
condense this information. Usually we notice users don't have the patience to read through all of this info to understand what they need to do to achieve something. The video is already there to explain things in detail. so for textual content instead of explaining every step in detail,  you can assume that the query already exists and then what the user should bind and where

Reply:
I agree in general, however in this particular case I think it might be best to keep these steps for **Total records**.

#### info-dump, sorry!

For most properties, a user will already have a query set up and the beginning steps aren't necessary, as you said. However for this one, I think it will be very rare for a user to have _already set up_ their own query specifically to count the number of records. And then, while basic `SELECT` queries are fairly common-knowledge, I think it will be more rare for a user to already know the syntax for using `SELECT COUNT(*)`.

---

More generally (kinda off-topic for this review comment), I also wanted to mention some thoughts I have about videos. They're super useful, though I try to use them strictly as supporting material and keep all of the information accessible via text. I'm a little hesitant to assume that users will have watched any given video, because:

- personally, videos are my last resort when using docs because they tend to take much longer to convey the information
- most of our videos aren't narrated and lack explanation for what/why we're doing (it's not usually hard to guess, but still)
- it's more difficult to find/skip in the video to rewatch a certain bit, than it is to read the step again
- we record in 720p and the appsmith code font is thin, making it kinda hard to read text
- users with spotty or unstable internet may have extra difficulty
- it's probably mostly safe to assume that our users wouldn't be using the appsmith platform if they are visually-impaired or require screen-readers, but I still try to make sure the text could work on its own if needed.